### DISCUSSION TITLE
[v2] Use Custom Editing Components freezes when clicking the edit row button

### DISCUSSION THREAD

**Discussion by yann-combarnous** [KevinVandy/mantine-react-table#368] [Q&A]
### mantine-react-table version

2.0.0-beta.5

### react & react-dom versions

18.3.1

### Describe the bug and the steps to reproduce it

**Steps to reproduce:**

When using the custom editing component documented [here](https://v2.mantine-react-table.com/docs/guides/editing#use-custom-editing-components), I just tried to use the documented example:
```
      Edit: ({ cell, table }) => (
        <MRT_EditCellTextInput cell={cell} key={cell.id} table={table} />
      ),
```

**Expected behavior:**
- when clicking on the row edit button, the edit modal should appear

**Actual behavior:**
- when clicking on the row edit button, the edit modal does not show up and page freezes without console error

### Minimal, Reproducible Example - (Optional, but Recommended)

Reproduced at [Code Sandbox](https://codesandbox.io/p/sandbox/mrt-edit-bug-75zfv3?file=%2Fsrc%2FApp.js%3A37%2C29&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clxd27efc00063p6swsahvimz%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clxd27efc00023p6syo730sz9%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clxd27efc00033p6sim3ots45%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clxd27efc00053p6stwfx5gzi%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B60.475899284419405%252C39.524100715580595%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clxd27efc00023p6syo730sz9%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxd27efc00013p6s2279nzw3%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clxd2q6x500023p6rboehpreh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A37%252C%2522startColumn%2522%253A29%252C%2522endLineNumber%2522%253A37%252C%2522endColumn%2522%253A29%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clxd27efc00023p6syo730sz9%2522%252C%2522activeTabId%2522%253A%2522clxd2q6x500023p6rboehpreh%2522%257D%252C%2522clxd27efc00053p6stwfx5gzi%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxd27efc00043p6sn3ca1zjh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clxd27efc00053p6stwfx5gzi%2522%252C%2522activeTabId%2522%253A%2522clxd27efc00043p6sn3ca1zjh%2522%257D%252C%2522clxd27efc00033p6sim3ots45%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clxd27efc00033p6sim3ots45%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

Note that by just removing the column's Edit option in column definition, the edit modal then shows up.

### Do you intend to try to help solve this bug with your own PR?

None

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by KevinVandy** [comment]
You didn't give your columns a stable reference

**Comment by yann-combarnous** [comment]
> You didn't give your columns a stable reference

Thanks, I added a stable reference in the sandbox, and still having the same issue.

**Comment by alessandrojcm** [comment]
@yann-combarnous Your custom edit component returns `<MRT_EditCellTextInput/>` which internally calls `column.Edit`, which in turn causes a render loop. Pass a custom component to the `Edit` prop of your column def and it should work.
